code/game/objects/items/devices/radio/intercom.dm

```diff
@@ -93,23 +89,28 @@
 
 	return canhear_range
 
-/obj/item/device/radio/intercom/Process()
-	if(((world.timeofday - last_tick) > 30) || ((world.timeofday - last_tick) < 0))
-		last_tick = world.timeofday
-
-		if(!src.loc)
-			on = FALSE
-		else
-			var/area/A = get_area(src)
-			if(!A)
-				on = FALSE
-			else
-				on = A.powered(EQUIP) // set "on" to the power status
-
-		if(!on)
-			icon_state = "intercom-p"
-		else
-			icon_state = "intercom"
+/obj/item/device/radio/intercom/proc/change_status()
+	on = linked_area.powered(EQUIP)
+	icon_state = on ? "intercom" : "intercom-p"
+
+/obj/item/device/radio/intercom/proc/loop_area_check()
+	var/area/target_area = get_area(src)
+	if(!target_area)
+		addtimer(CALLBACK(src, .proc/loop_area_check), 30 SECONDS)
+		return FALSE
+	if(!target_area.apc)
+		addtimer(CALLBACK(src, .proc/loop_area_check), 30 SECONDS) // We don't proces if there is no APC , no point in doing so is there ?
+		return FALSE
```

Comment:
```suggestion
	if(!target_area || !target_area.apc)
		addtimer(CALLBACK(src, .proc/loop_area_check), 30 SECONDS)
		return FALSE
```

Reply:
runtime on null area , not a good idea.

Reply:
👀 
listen to this poor old man.

Reply:
> however i get the idea and ?. should do.

I already made my suggestion, if you want to implement something else that's fine.